### ISSUE TITLE
Detected I/O inside the event loop

### ISSUE THREAD

**Issue by frankverheyen** [herikw/home-assistant-custom-components#24]
This is a regular returning message in the home assistant (hassio) log
Detected I/O inside the event loop. This is causing stability issues. Please report issue to the custom component author for atagone doing I/O at custom_components/atagone/atagoneapi.py, line 94: with urlopen(req, timeout=60) as result:

I am receiving data from ATAG One, but I do not like the error :-)
Atagone version Installed: v0.1.5
Home Assistant 0.109.6
Supervisor version 222
Installed on rpi4

Any ideas? Thanks

**Comment by RezzZ** [comment]
same here. probably because the atagone thermostat has such a crapy wifi connection..

**Comment by reharmsen** [comment]
You might want to consider using the 'official' component that is offered in the Core that was introduced with 0.109: https://www.home-assistant.io/integrations/atag/

And perhaps suggest changes when you are missing features.

**Comment by herikw** [comment]
There are also advantages using components outside the HA core. You don't have to wait for a bug fix until home-assistant core has been updated and released. I think HACS is a much better solution to only add components that you really need. This makes the whole surface a lot smaller and easier to update and maintain. But they are changing how the core functions so frequently that it is hard to follow the developments of HA. But the same counts for the core components. A lot are not up-to-date with the latest changes in the Core. 
Anyway, the "Detected I/O inside the event loop" is fixed in my latest release of the AtagOne Component.
Btw. It's not the crapy wifi connection that stops the communication to the ATAG One on occasion, but the ATAG One can not handle the amount of connection requests that are made. This is causing timeouts. (each sensor update is basically a connection)
